pkg/ephemeral-disk/ephemeral-disk.go

```diff
@@ -36,7 +36,7 @@ const (
 
 type EphemeralDiskCreatorInterface interface {
 	CreateBackedImageForVolume(volume v1.Volume, backingFile string) error
-	CreateEphemeralImages(vmi *v1.VirtualMachineInstance) error
+	CreateEphemeralImages(vmi *v1.VirtualMachineInstance, isBlockVolume map[string]bool) error
```

Comment:
I'm not entirely fond of having that map[string]bool parameter on this interface method.
Please consider if it makes sense to have the `ephemeralDiskCreator` itself be configured with some function that determines if a volume is a block volume (maybe hook it up with the `ConverterContext.IsBlockPVC`, or with the `types.IsPVCBlockFromStore` function?). I'm not sure this will result in a cleaner design, but let's try that.

Reply:
Originally I did pass `ConverterContext.IsBlockPVC` through, but that required changing additional functions. My aim was to change the least number of function prototypes, and, as best as I could, keep function changes to those that are only relevant to ephemeral volumes. To pass the ConverterContext requires changing the function prototype for `preStartHook`.

Additionally, correct me in I'm wrong, but in this context (the virt-launcher) there isn't and shouldn't be access to the cache.store required for `types.IsPVCBlockFromStore` to function. 

When I attempted to do as you recommended I came up with two solutions:
1) Pass the `converter` or `converterContext` into the `CreateEphemeralImage` function and use it to decide which volumes were block devices
2) Pass the `IsBlockDeviceVolumeFunc` into the `ephemeralDiskCreator` struct on instantiation in `virt-launcher.go`. (It can't be imported straight into `ephemeral-disk.go` as that would cause a dependency loop.

Both turned out to be exceptionally unclean. I ended up moving `isDevEphemeralBackingSource` out into a separate helper function as requested. I moved it into the ephemeral disk `utils.go`, and instead pass the domain to `CreateEphemeralImage` which I think is a cleaner solution.

Reply:
> I ended up moving isDevEphemeralBackingSource out into a separate helper function as requested. I moved it into the ephemeral disk utils.go, and instead pass the domain to CreateEphemeralImage which I think is a cleaner solution.

That indeed turned out notably cleaner. Thanks! :-)